django/contrib/admindocs/views.py

```diff
@@ -353,8 +355,14 @@ def get_context_data(self, **kwargs):
             # Non-trivial TEMPLATES settings aren't supported (#24125).
             pass
         else:
-            # This doesn't account for template loaders (#24128).
-            for index, directory in enumerate(default_engine.dirs):
+            directories = set(default_engine.dirs)
+            for loader in default_engine.template_loaders:
+                if isinstance(loader, Loader):
+                    directories.update(loader.get_dirs())
+                elif isinstance(loader, Cached_Loader):
```

Comment:
I agree, when I was making that change I was thinking about it, but have to look into the Cached_Loader, maybe easiest would be to add the get_dirs(), the test should still work and then there is no need to check which object type it is.

I can look into it tomorrow to see if I am able to write the change.